compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1414,7 +1414,11 @@ impl<'a> Parser<'a> {
 
                 Ok((Some(vr), TrailingToken::MaybeComma))
             },
-        )
+        );
+        if let Err(ref mut err) = result_tt {
+            err.help("enum variants can be `Variant` or `Variant = <integer>` or `Variant(Type)` or `Variant { field: Type }`");
+        }
+        return result_tt;
```

Comment:
```suggestion
        result_tt
```
though it's possible the `collect_tokens_trailing_token` can be chained with a `map_err` or an `inspect_err` call?

Reply:
yes collect_tokens_trailing_token can be chained with a map_err